src/utils/cors/handler.ts

```diff
@@ -0,0 +1,24 @@
+import { defineEventHandler } from "../../event";
+import { sendNoContent } from "../response";
+import {
+  resolveCorsOptions,
+  appendCorsPreflightHeaders,
+  appendCorsActualRequestHeaders,
+  isPreflightRequest,
+} from "./utils";
+import type { CorsOptions } from "./types";
+
+export function handleCors(options: CorsOptions) {
+  const {
+    preflight: { statusCode },
+  } = resolveCorsOptions(options);
+
+  return defineEventHandler((event) => {
```

Comment:
handle shouldn't create a wrapper handler but directly accept `event` to append required headers.

Reply:
Fixed: c1eef73